src/controllers/beneficial.owner.type.controller.ts

```diff
@@ -57,14 +41,30 @@ export const postSubmit = async (req: Request, res: Response) => {
   const appData: ApplicationData = await getApplicationData(req.session);
   const requiresTrusts: boolean = checkEntityRequiresTrusts(appData);
   let nextPageUrl = config.CHECK_YOUR_ANSWERS_URL;
+
+  const errors = await getValidationErrors(appData, req);
+
+  if (errors.length) {
+    return await getPageRender(req, res, formatValidationError(errors));
+  }
+
   if (isActiveFeature(config.FEATURE_FLAG_ENABLE_REDIS_REMOVAL)) {
     nextPageUrl = getUrlWithParamsToPath(config.CHECK_YOUR_ANSWERS_WITH_PARAMS_URL, req);
   }
+
   if (requiresTrusts) {
     nextPageUrl = isActiveFeature(config.FEATURE_FLAG_ENABLE_TRUSTS_WEB)
       ? getTrustLandingUrl(appData, req)
       : config.TRUST_INFO_URL;
   }
+
+  if (isActiveFeature(config.FEATURE_FLAG_ENABLE_PROPERTY_OR_LAND_OWNER_NOC)) {
+    console.debug("Removing old NOCs");
```

Comment:
we should use logger instead of console, there should be a logger.debugRequest that will also log the request id to help identify it in the logs.

Reply:
changed to logger.debugRequest